### ISSUE TITLE
Implicit joined fields are not available as breakouts in Models

### ISSUE THREAD

**Issue by luizarakaki** [metabase/metabase#39657]
### Describe the bug

Implicit joined fields are not available as breakouts in Models
[Slack thread](https://metaboat.slack.com/archives/C04DN5VRQM6/p1709660912019889)

### To Reproduce

1. New Model > Sample database > Orders
2. Implicit joined fields are available for filters and aggregations, but not for breakouts

[Stats reproduction](https://stats.metabase.com/model/4520-orders-model/notebook)

Filter:
![image](https://github.com/metabase/metabase/assets/32561114/df607c7a-dc15-4878-a933-8ef0248bc865)

Breakout:
![image](https://github.com/metabase/metabase/assets/32561114/f7b83b61-e2d6-4d16-a73a-4f8e00e59f3f)

### Expected behavior

Implicit joined field are available for breakouts

### Information about your Metabase installation

```JSON
Master
```

### Severity

P1

**Comment by bshepherdson** [comment]
Some context:
- There are options for controlling what set of columns to show in various contexts.
- One of these is `:include-implicitly-joinable-for-source-card?`, which does what it says on the tin.
- That option defaults to `true` but is set to `false` for:
    - `breakoutable-columns`
    - suggesting join conditions
    - `orderable-columns`

That last actually suggests the same thing applies to order-by as well. Checking quickly reveals confirms this guess. Implicit joins are allowed for filters and aggregations, not for order-by or breakouts.

I think this bug arises from conflating plain source cards with models. We don't want to follow implicit joins on source cards, but we do on models, and this option is not specific enough.

On `suggested-join-conditions`, we don't want to include any implicitly joined columns at all. That code also sets `:include-implicitly-joinable? false` which actually makes `:include-implicitly-joinable-for-source-card?` redundant.

To summarize, here's the current state of where implicit joins are allowed and where they're not:

| Context | Vanilla `source-card` | Model |
| :-- | :-- | :-- |
| Suggested join conditions | :x: | :x: |
| Filters | :heavy_check_mark: | :heavy_check_mark: |
| Aggregations | :heavy_check_mark: | :heavy_check_mark: |
| Custom columns | :heavy_check_mark: | :heavy_check_mark: |
| Breakouts | :x: | :x: |
| Orders | :x: | :x: |

@luizarakaki can you give me the same table but showing what we want instead of what we have now? I want to confirm the intended behavior all around.
